Bookdetails

riaars opened the pull request.

Commit:
bookdetails

Commit:
book details

Commit:
book details

bt-s committed:
Clean up star.svg

Right now, this symbol is not used. Do we plan on using it in the
future?

bt-s committed:
Clean BookDetail

The current implementation of the getStars() function does not work
properly. I added the FA star components, since this works easier
than having to define the HTML yourself.

bt-s committed:
Clean up BookDetail stylesheet

bt-s committed:
Lint Rating component

bt-s committed:
Fix linting errors

bt-s committed:
Merge branch 'master' into bookdetails

bt-s committed:
Move BookDetail component to a page

bt-s commented on src/assets/star.svg:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
This file is not used yet.

bt-s replied:
This file is not used yet.

bt-s commented on src/components/BookDetail/_BookDetail.scss:
```diff
@@ -0,0 +1,124 @@
+.book-details-container {
```
I have made a couple of changes:

- Since left-right is viewport dependend, let's try to refrain from this.
- Instead of using @media for media queries, we can use the @include media-breakpoint(<size>) mixin.

This is a good start, but most of the page is not responsive yet.
This is fine for now, but should be changed duely.

bt-s replied:
I have made a couple of changes:

- Since left-right is viewport dependend, let's try to refrain from this.
- Instead of using @media for media queries, we can use the @include media-breakpoint() mixin.

This is a good start, but most of the page is not responsive yet.
This is fine for now, but should be changed duely.

bt-s commented on src/components/Rating/Rating.js:
```diff
@@ -0,0 +1,71 @@
+class StarRating extends HTMLElement {
```
I am not too sure about what this code does.
It would be nice if we could rewrite it to a React component, preferably writing it as a functional component using React hooks.

bt-s replied:
I am not too sure about what this code does.
It would be nice if we could rewrite it to a React component, preferably writing it as a functional component using React hooks.

bt-s commented on src/pages/BookDetail.js:
```diff
@@ -0,0 +1,81 @@
+import React from 'react';
```
I have imported the FA icons directly from the FA library, which is a bit more user-friendly.
However, the getStars() function does not work properly yet.

Maybe we should finish this functionality before we merge it with master.

Envl replied:
@riaars I've fixed this.
What else do you need to do? And what kind of help may I offer?

bt-s replied:
@Envl where can we see that you have fixed this?

Envl replied:
@bt-s I've pushed to this branch. Just check it at /detail page.

bt-s commented on package.json:
```diff
@@ -14,6 +14,7 @@
     "google-maps-react": "^2.0.2",
```
Please be really careful with running a command like $ npm update, since this usually leads to a lot of merge conflicts in package-lock.json (which it did this time).

bt-s reviewed (changes requested):
This is a nice start @riaars! I have made some changes to the code, such that it is compatible with the current master branch. Please have a look at the comments that I have made and see the separate commits that I have pushed for further remarks.

I am sure that this is going to be a nice page, but I would suggest that we first try to fix its functionality (i.e. the rating component etc.) before merging it with the master branch.

Envl committed:
fix logic issue in getStars()
I also disabled the empty stars because I couldn't find a empty star in fontawesome

bt-s reviewed (dismissed):
Looks better!

Envl reviewed (approved):
I think it's now ok to merge for mid-project submission.

Commit:
fix

Commit:
responsive layout
